lib/config/validation.ts

```diff
@@ -562,18 +550,13 @@ export async function validateConfig(
                 }
               }
             }
-            if (
-              [
-                'matchPackagePatterns',
-                'excludePackagePatterns',
-                'matchDepPatterns',
-                'excludeDepPatterns',
-              ].includes(key)
-            ) {
+            if (['matchPackageNames', 'matchDepNames'].includes(key)) {
+              const startPattern = regEx(/!?\//);
+              const endPattern = regEx(/\/g?i?$/);
               for (const pattern of val as string[]) {
-                if (pattern !== '*') {
+                if (pattern.match(startPattern) && pattern.match(endPattern)) {
                   try {
-                    regEx(pattern);
+                    regEx(pattern.replace(startPattern, '/'));
```

Comment:
why the replace? don't we need to remove the end pattern too?

Reply:
`regEx() doesn't recognize our `!/..../` syntax but should be fine with the end syntax. Hence I replace any leading `!/` with `/` but leave the trailing as-is